src/domain/storage/repositories/StorageClient.ts

```diff
@@ -8,6 +8,25 @@ export interface StorageClientConstructor {
     new (instance: Instance): StorageClient;
 }
 
+export type AccessRule = string;
+
+export interface SharingRule {
+    id: string;
+    access: AccessRule;
+    displayName: string;
+}
```

Comment:
We already have an entity called ``SharingSetting`` used in the app with essentially the same behavior, I would personally re-use it (src/domain/common/entities/SharingSetting).

```ts
export interface SharingSetting {
    access: string;
    displayName: string;
    id: string;
    name?: string;
}
```

In other apps where we handle sharing settings, we even have schema validation already created for it and some helper methods to check permissions abstracted into helper files.

Reply:
ShareSettings is only a part of the /api/sharing network response for sharing and shareRef domain entity is not equal to network response in this case. In clean architecture is normal these cases.  I personally would not re-use it, so SharingObject and SharingRule can be moved to d2api in the future when d2api implements this feature.